### ISSUE TITLE
Multiple declarations not flagged

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#1243]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

some_func ()
{
        local verbose=1
        declare verbose
        verbose=0
        echo $verbose
}

```

#### Here's what shellcheck currently says:
```
Nothing
```
#### Here's what I wanted or expected to see:

SCXXXX: 'verbose' declared again at line 3
